src/TpetraLinearSystem.C

```diff
@@ -245,17 +244,62 @@ void TpetraLinearSystem::beginLinearSystemConstruction()
   std::vector<stk::mesh::Entity>::iterator iter = std::unique(owned_nodes.begin(), owned_nodes.end(), CompareEntityEqualById(bulkData, realm_.naluGlobalId_));
   owned_nodes.erase(iter, owned_nodes.end());
 
+  ThrowRequire((int)maxOwnedRowId_ == (int)(owned_nodes.size()*numDof_)); // if this is never tripped, then the above unique is useless. !!!
+
```

Comment:
So is it useless or not? If it's useless, remove it. If not useless, remove this comment.

Reply:
I'll remove it.